src/App/Fossa/Config/ConfigFile.hs

```diff
@@ -90,18 +94,23 @@ resolveConfigFile base path = do
   loc <- resolveLocation base path
   case loc of
     DefaultConfigLocation dir -> do
-      let realpath = dir </> defaultConfigFileName
-      exists <- doesFileExist realpath
-      if not exists
-        then -- File missing and not requested
-          pure Nothing
-        else do
-          configFile <- readContentsYaml realpath
+      possibleConfigFilePaths <-
+        mapM
+          ( \configFileName -> do
+              let configFilePath = dir </> configFileName
+              exists <- doesFileExist configFilePath
+              pure $ if exists then Just configFilePath else Nothing
```

Comment:
This monad doesn't have `Maybe` as it's failure monad case does it?  If so you could just write `guard exists $> configFilePath`

Reply:
I don't quite want to use `guard` here because I want the result to be `[Maybe a]` not `[a]` because `asum` already conveniently gives me `Nothing` in the empty case.